src/v2/guide/typescript.md

```diff
@@ -139,3 +139,38 @@ var vm = new Vue({
   myOption: 'Hello'
 })
 ```
+
+## Annotating Return Type
+
+Vue's type definition is prone to [circular reference](https://github.com/Microsoft/TypeScript/issues/12846#issuecomment-270296195) of `this`.
+Annotating return type is highly recommended to avoid such error.
```

Comment:
Consider

```
Because of the circular nature of Vue's declaration files, TypeScript may have difficulties inferring the types of certain methods.
For this reason, you may need to annotate the return type on `computed` methods.
```